petpal/applications/views.py

```diff
@@ -43,7 +252,66 @@ def pet_applications_list_view(request, pet_id):
 @api_view(['GET', 'PUT'])
 def application_detail_view(request, app_id):
     if request.method == 'GET':
-        return Response({'msg': 'application detail'}, status=status.HTTP_200_OK)
+        try:
+            app = Application.objects.get(pk=app_id)
+        except Application.DoesNotExist:
+            return Response({'error':'app id not found'}, status=status.HTTP_400_BAD_REQUEST)
+
+        if app.seeker !=  request.user and app.shelter != request.user:
+            return Response({'error':'unauthorized'}, status=status.HTTP_401_UNAUTHORIZED)
+
+        serialized_app = ApplicationSerializer(app)
+        return JsonResponse({'data': serialized_app.data}, status=status.HTTP_200_OK)
+
+    elif request.method == 'PUT':
+        data = {
+            "status" : request.data.get('status', ''),
+        }
+
+        put_app = Application.objects.get(pk=app_id)
+       
+
+        new_status = data["status"]
+
+        if new_status == put_app.status:
+            return Response({'error':'status did not change'}, status=status.HTTP_400_BAD_REQUEST)
+        
+        curr_user = PetPalUser.objects.get(pk=request.user.id)
+
+        if curr_user.role == PetPalUser.Role.SHELTER:
+            if put_app.shelter != curr_user:
+                return Response({'error':'unauthorized'}, status=status.HTTP_401_UNAUTHORIZED)
+
+            if new_status not in ['A', 'D']:
+                return Response({'error':f'invalid status: {new_status}'}, status=status.HTTP_400_BAD_REQUEST)
+
+        elif curr_user.role == PetPalUser.Role.SEEKER:
+            if put_app.seeker != curr_user:
+                return Response({'error':'unauthorized'}, status=status.HTTP_401_UNAUTHORIZED)
+
+            if new_status not in ['W']:
+                return Response({'error':f'invalid status: {new_status}'}, status=status.HTTP_400_BAD_REQUEST)
+        else:
+            return Response({'error':f'invalid user role: {curr_user.role}'}, status=status.HTTP_400_BAD_REQUEST)
+
+        put_app.status = new_status
+        put_app.save()
+
+        # notify user about status update
+        if put_app.seeker.is_notif_status:
```

Comment:
The logic to add recipients to the notification should be different, if a user updated, it should send to shelter, and if shelter updated, it should send to user

Reply:
Otherwise looks good